frontend/src/store/shoot/index.js

```diff
@@ -440,161 +459,199 @@ export const useShootStore = defineStore('shoot', () => {
         }
-      })
-
-      addedShootKeys.forEach(addedShootKey => {
-        const addedShoot = shoots[addedShootKey]
-        const uid = addedShoot.metadata.uid
-        delete state.staleShoots[uid]
-      })
-    }
-
-    state.shoots = shoots
-    updateFilteredShoots()
+        for (const uid of difference(newUids, oldUids)) {
+          delete state.staleShoots[uid]
+        }
+      }
+      state.shoots = shoots
+    })
   }
 
-  function receiveInfo ({ namespace, name, info }) {
-    const item = findItem(state)({ namespace, name })
-    if (item !== undefined) {
-      item.info = info
+  function setSubscriptionEventHandler (state, func, throttleDelay) {
+    if (typeof state.subscriptionEventHandler?.cancel === 'function') {
+      state.subscriptionEventHandler.cancel()
     }
+    shootEvents.clear()
+    if (throttleDelay > 0) {
+      func = throttle(func, throttleDelay)
+    }
+    state.subscriptionEventHandler = markRaw(func)
   }
 
-  function clear () {
-    state.shoots = {}
-    state.staleShoots = {}
+  function unsetSubscriptionEventHandler (state) {
+    if (typeof state.subscriptionEventHandler?.cancel === 'function') {
+      state.subscriptionEventHandler.cancel()
+    }
+    shootEvents.clear()
+    state.subscriptionEventHandler = undefined
   }
 
-  function clearStaleShoots () {
-    state.staleShoots = {}
+  async function openSubscription (value, throttleDelay) {
+    const shootStore = this
+    shootStore.$patch(({ state }) => {
+      setSubscriptionState(state, constants.OPENING)
+      setSubscriptionEventHandler(state, handleEvents, throttleDelay)
+    })
+    try {
+      await socketStore.emitSubscribe(value)
+      setSubscriptionState(state, constants.OPEN)
+    } catch (err) {
+      setSubscriptionError(state, err)
+    }
   }
 
-  function openSubscription (options) {
-    state.subscriptionState = constants.OPENING
-    state.subscriptionError = null
-    socketStore.emitSubscribe(options)
+  async function closeSubscription () {
+    const shootStore = this
+    shootStore.$patch(({ state }) => {
+      state.subscription = null
+      setSubscriptionState(state, constants.CLOSING)
+      unsetSubscriptionEventHandler(state)
+    })
+    try {
+      await socketStore.emitUnsubscribe()
+      setSubscriptionState(state, constants.CLOSED)
+    } catch (err) {
+      setSubscriptionError(state, err)
+    }
   }
 
-  function closeSubscription () {
-    state.subscriptionState = constants.CLOSING
-    state.subscriptionError = null
-    state.subscription = null
-    socketStore.emitUnsubscribe()
+  async function handleEvents (shootStore) {
+    const events = Array.from(shootEvents.values())
+    shootEvents.clear()
+    const uids = []
+    const deletedUids = []
+    for (const { type, uid } of events) {
+      if (type === 'DELETED') {
+        deletedUids.push(uid)
+      } else {
+        uids.push(uid)
+      }
+    }
+    try {
+      const items = await socketStore.synchronize(uids)
```

Comment:
idea: do not run syncronize when tab is in background

Reply:
Synchronization is stop when the tab is in the backendground now. The next step would be to pause and resume the subscription. I have create the issue https://github.com/gardener/dashboard/issues/1653 for that.